static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -24,6 +38,202 @@ interface BaseDraggableTabListProps extends DraggableTabListProps {
+  const addViewTempTabRef = useRef<HTMLDivElement>(null);
+  const [tabElements, setTabElements] = useState<Array<HTMLDivElement | null>>([]);
+  const {width: outerWidth} = useDimensions({elementRef: outerRef});
+  const {width: addViewTempTabWidth} = useDimensions({elementRef: addViewTempTabRef});
+  const tabsDimensions = useDimensionsMultiple({elements: tabElements});
+
+  const overflowingTabs = useMemo(() => {
+    const availableWidth = outerWidth - addViewTempTabWidth;
+
+    let totalWidth = 0;
+    const overflowing: Node<DraggableTabListItemProps>[] = [];
+
+    for (let i = 0; i < tabsDimensions.length; i++) {
+      totalWidth += tabsDimensions[i].width + 1; // 1 extra pixel for the divider
+      if (totalWidth > availableWidth + 1) {
+        overflowing.push(persistentTabs[i]);
+      }
+    }
+
+    return overflowing.filter(defined);
+  }, [outerWidth, addViewTempTabWidth, persistentTabs, tabsDimensions]);
+
+  return {
+    overflowingTabs,
+    setTabElements,
+    outerRef,
+    addViewTempTabRef,
+    persistentTabs,
+  };
+}
+
+function OverflowMenu({
+  state,
+  overflowTabs,
+}: {
+  overflowTabs: Node<DraggableTabListItemProps>[];
+  state: TabListState<any>;
+}) {
+  const options = useMemo(() => {
+    return overflowTabs.map(tab => {
+      return {
+        value: tab.key,
+        label: tab.textValue,
+        textValue: tab.textValue,
+      };
+    });
+  }, [overflowTabs]);
+
+  return (
+    <CompactSelect
+      options={options}
+      multiple={false}
+      value={state.selectionManager.firstSelectedKey?.toString()}
+      onChange={opt => state.setSelectedKey(opt.value)}
+      position="bottom-end"
+      size="sm"
+      offset={4}
+      trigger={triggerProps => (
+        <OverflowMenuTrigger
+          {...triggerProps}
+          size="sm"
+          borderless
+          showChevron={false}
+          icon={<IconEllipsis />}
+          aria-label={t('More tabs')}
+        />
+      )}
+    />
+  );
+}
+
+function Tabs({
+  orientation,
+  ariaProps,
+  state,
+  className,
+  onReorder,
+  tabVariant,
+  setTabRefs,
+  tabs,
+  overflowingTabs,
+}: {
+  ariaProps: AriaTabListOptions<DraggableTabListItemProps>;
+  onReorder: (newOrder: Node<DraggableTabListItemProps>[]) => void;
+  orientation: 'horizontal' | 'vertical';
+  overflowingTabs: Node<DraggableTabListItemProps>[];
+  setTabRefs: Dispatch<SetStateAction<Array<HTMLDivElement | null>>>;
+  state: TabListState<DraggableTabListItemProps>;
+  tabs: Node<DraggableTabListItemProps>[];
+  className?: string;
+  disabled?: boolean;
+  onChange?: (key: string | number) => void;
+  tabVariant?: BaseTabProps['variant'];
+  value?: string | number;
+}) {
+  const tabListRef = useRef<HTMLUListElement>(null);
+  const {tabListProps} = useTabList({orientation, ...ariaProps}, state, tabListRef);
+
+  const values = useMemo(() => [...state.collection], [state.collection]);
+
+  const isDragging = useIsDragging(tabListRef);
```

Comment:
nit: rather than creating this hook, I think you could use Framer motion's `onDragStart` and `onDragEnd` on each tab to update an `isDragging` state

Reply:
You're totally right! That's much better